relay-server/src/actors/project_local.rs

```diff
@@ -136,31 +175,18 @@ fn load_local_states(projects_path: &Path) -> io::Result<HashMap<ProjectKey, Arc
     Ok(states)
 }
 
-fn poll_local_states(
-    manager: Addr<LocalProjectSource>,
-    config: Arc<Config>,
-) -> impl Future<Item = (), Error = ()> {
-    let (sender, receiver) = oneshot::channel();
-
-    let _ = thread::spawn(move || {
-        let path = config.project_configs_path();
-        let mut sender = Some(sender);
-
-        loop {
-            match load_local_states(&path) {
-                Ok(states) => {
-                    manager.do_send(UpdateLocalStates { states });
-                    sender.take().map(|sender| sender.send(()).ok());
-                }
-                Err(error) => relay_log::error!(
-                    "failed to load static project configs: {}",
-                    LogError(&error)
-                ),
-            }
-
-            thread::sleep(config.local_cache_interval());
+async fn poll_local_states(
+    path: &Path,
+    tx: &mut mpsc::Sender<HashMap<ProjectKey, Arc<ProjectState>>>,
+) {
+    let states = load_local_states(path).await;
+    match states {
+        Ok(states) => {
+            let _ = tx.send(states).await;
```

Comment:
Should the error be handled here which can happen when the channel is closed? Or maybe just log it?  This also comes to other instances of `send` where  (if) the result is returned.